services/py-api/src/service/hackathon_service.py

```diff
@@ -69,6 +69,22 @@ async def create_random_participant(
 
         # As when first created, the random participant is not assigned to a team we return the team as None
         return Ok((result.ok_value, None))
+    
+    async def create_invite_link_participant(
+            self, input_data: ParticipantRequestBody
+    ) -> Result[Tuple[Participant, Team], DuplicateEmailError | Exception]:
+        
+        participant_result = await self._participant_repo.create(input_data)
+        if is_err(participant_result):
+            return participant_result
+
+        # Assign the team whose name matches that from input_data.team_name
+        team_result = await self._team_repo.fetch_team_by_name(input_data.team_name)
+        if is_err(team_result):
+            return team_result
```

Comment:
I dont think we need to return the team too, since we are not adding a new team or updating a new one, so you can omit this.

Reply:
I will use it to get the team id but tell me if there is still a way to get rid of it